parsing/parsetree.mli

```diff
@@ -471,6 +472,11 @@ and function_param =
       variables [a], [b], and [c] in the source code.
   *)
 
+and function_param =
+  { pparam_loc : Location.t;
+    pparam_desc : function_param_desc;
+  }
```

Comment:
I am not sure whether one should rather use this or something like

```ocaml
and function_param = function_param_desc loc
```

which is isomorphic.

Today in the parsetree codebase:
- We only use the `_desc / _loc` patterns for records that have other metadata fields than just locations (typically attributes, sometimes more). This suggests using ` loc .
- We only use `foo loc` for "external" foos that are themselves defined in Parsetree, typically `string` or `label` or `string option`. This suggests using a `_desc / _loc` record.

Any opinion?

Reply:
My opinion is that we should use a dedicated record for the second reason you gave. As another point in that direction, the two fields of `loc` are `loc` and `txt`, and `txt` does not seem like a very good name for a field that contains the parsed function parameter.